blockchain/My-Wallet-V3-iOS
# feat(KYC): IOS-1174 use new mechanism for verifying phone number during KYC

## Objective

Use the new mechanism for verifying phone number during KYC.

## Description

The previous endpoints that were used to verify a user's phone number (i.e. `/users/current/mobile`) will soon be deprecated in favor of the following flow: (1) updating the user's wallet via wallet settings, followed by (2) obtaining a signed retail JWT token from the wallet settings and sending that to Nabu to update the user's mobile phone number. The reason this method is preferred is because this updates both the wallet and the user's phone number.

**Other changes:**
Since this means the KYC feature needs to interface with `Wallet.m/.h`, I did a bit of refactoring in the `changeMobileNumber`/`verifyMobileNumber` functions so that a success/error block can be passed in directly as opposed to using a global delegate/handler.

## How to Test

Pull in these changes, launch KYC using the debug side menu option, and go through the KYC flow. Note that this still requires a unique phone number to be used for the user to continue. 

Currently, the dev environment does not allow 2 users to have the same phone number but this will be changed soon (supposedly tomorrow).

Since I also changed some code in the mobile verification flow in settings, please make sure to test those areas as well. I did some manual testing myself but would like to get an extra pair of 👀 there, too.

## Screenshot/Design assessment

N/A

## Merge Checklist
